modules/RTC/JitsiLocalTrack.js

```diff
@@ -15,696 +15,697 @@ import VideoType from '../../service/RTC/VideoType';
+        Statistics.analytics.sendEvent(eventName);
+        const log = { name: eventName };
 
-/* eslint-enable max-params */
+        if (this.isAudioTrack()) {
+            log.isReceivingData = this._isReceivingData();
+        }
+        Statistics.sendLog(JSON.stringify(log));
+    }
 
-JitsiLocalTrack.prototype = Object.create(JitsiTrack.prototype);
-JitsiLocalTrack.prototype.constructor = JitsiLocalTrack;
+    /**
+     * Sets real device ID by comparing track information with device
+     * information. This is temporary solution until getConstraints() method
+     * will be implemented in browsers.
+     * @param {MediaDeviceInfo[]} devices - list of devices obtained from
+     *  enumerateDevices() call
+     */
+    _setRealDeviceIdFromDeviceList(devices) {
+        const track = this.getTrack();
 
-/**
- * Returns if associated MediaStreamTrack is in the 'ended' state
- * @returns {boolean}
- */
-JitsiLocalTrack.prototype.isEnded = function() {
-    return this.getTrack().readyState === 'ended' || this._trackEnded;
-};
+        // FIXME for temasys video track, label refers to id not the actual
+        // device
+        const device = devices.find(
+            d => d.kind === `${track.kind}input` && d.label === track.label);
 
-/**
- * Sets handlers to the MediaStreamTrack object that will detect camera issues.
- */
-JitsiLocalTrack.prototype._initNoDataFromSourceHandlers = function() {
-    if (this.isVideoTrack() && this.videoType === VideoType.CAMERA) {
-        const _onNoDataFromSourceError
-            = this._onNoDataFromSourceError.bind(this);
-
-        this._setHandler('track_mute', () => {
-            if (this._checkForCameraIssues()) {
-                const now = window.performance.now();
-
-                this._noDataFromSourceTimeout
-                    = setTimeout(_onNoDataFromSourceError, 3000);
-                this._setHandler('track_unmute', () => {
-                    this._clearNoDataFromSourceMuteResources();
-                    Statistics.sendEventToAll(
-                        `${this.getType()}.track_unmute`,
-                        { value: window.performance.now() - now });
-                });
-            }
-        });
-        this._setHandler('track_ended', _onNoDataFromSourceError);
+        if (device) {
+            this._realDeviceId = device.deviceId;
+        }
     }
-};
 
-/**
- * Clears all timeouts and handlers set on MediaStreamTrack mute event.
- * FIXME: Change the name of the method with better one.
- */
-JitsiLocalTrack.prototype._clearNoDataFromSourceMuteResources = function() {
-    if (this._noDataFromSourceTimeout) {
-        clearTimeout(this._noDataFromSourceTimeout);
-        this._noDataFromSourceTimeout = null;
+    /**
+     * Sets the stream property of JitsiLocalTrack object and sets all stored
+     * handlers to it.
+     * @param {MediaStream} stream the new stream.
+     */
+    _setStream(stream) {
+        JitsiTrack.prototype._setStream.call(this, stream);
```

Comment:
I think you're also changing the functions a bit for es6. Am I correct assuming so? Could this be changed to a super._setStream call?